src/test/java/picard/sam/ReorderSamTest.java

```diff
@@ -0,0 +1,194 @@
+package picard.sam;
+
+import htsjdk.samtools.*;
+import htsjdk.variant.utils.SAMSequenceDictionaryExtractor;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.util.SequenceDictionaryUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Created by farjoun on 1/12/18.
+ */
+public class ReorderSamTest extends CommandLineProgramTest {
+
+    @Override
+    public String getCommandLineProgramName() {
+        return ReorderSam.class.getSimpleName();
+    }
+
+    @DataProvider(name = "testDictionaryData")
+    public Iterator<Object[]> testDictionaryData() throws IOException {
+        List<Object[]> tests = new ArrayList<>();
+
+        final SAMRecordSetBuilder setBuilder = new SAMRecordSetBuilder();
+
+        setBuilder.addPair("pair1", setBuilder.getHeader().getSequenceIndex("chr1"), 1, 100);
+        setBuilder.addPair("pair2", setBuilder.getHeader().getSequenceIndex("chr2"), 1, 100);
+        setBuilder.addPair("pair3", setBuilder.getHeader().getSequenceIndex("chr3"), 1, 100);
+        setBuilder.addPair("pair4", setBuilder.getHeader().getSequenceIndex("chr4"), 1, 100);
+        setBuilder.addPair("pair5", setBuilder.getHeader().getSequenceIndex("chr5"), 1, 100);
+
+        setBuilder.addUnmappedFragment("unmapped");
+        setBuilder.addUnmappedPair("unmapped");
+
+        setBuilder.addPair("split_pair1",
+                setBuilder.getHeader().getSequenceIndex("chr1"),
+                setBuilder.getHeader().getSequenceIndex("chr6"),
+                100, 200,
+                false, false, "36M",
+                "36M", true, false, false,
+                false, 30);
+
+        setBuilder.addPair("split_pair2",
+                setBuilder.getHeader().getSequenceIndex("chr1"),
+                setBuilder.getHeader().getSequenceIndex("chr6"),
+                100, 200,
+                false, true, "36M",
+                "*", true, false, false,
+                false, 30);
+
+        { // same dictionary, but different order
+            final File dictionary = File.createTempFile("reorder-shuffle", ".dict");
+            dictionary.deleteOnExit();
+
+            final List<SAMSequenceRecord> sequences = new ArrayList<>(new SAMRecordSetBuilder().getHeader().getSequenceDictionary().getSequences());
+            Collections.shuffle(sequences, new Random(42));
```

Comment:
Ah, I guess if you push `createTempFile` into the test body then you can't write to the file here. I think you can solve that problem and DRY up even more if you pass a `Consumer<List<SAMSequenceRecord>>` to the test, instead of the final `List<SAMSequenceRecord>`. Then you can move the repeated `new ArrayList<>(...)` and `writeDictionary` calls into the body of the test too.

Reply:
done.